packages/ra-core/src/i18n/useTranslate.ts

```diff
@@ -21,9 +22,17 @@ import { TranslationContext } from './TranslationContext';
  *     return <MenuItem>{translate('settings')}</MenuItem>;
  * }
  */
-const useTranslate = () => {
-    const { translate } = useContext(TranslationContext);
-    return translate;
+const useTranslate = (): Translate => {
+    const { i18nProvider, locale } = useContext(TranslationContext);
+    const translate = useCallback(
+        (key: string, options?: any) =>
+            i18nProvider(I18N_TRANSLATE, { key, options }) as string,
```

Comment:
We are not accepting promises at all for this call ?

Reply:
Because if translate is asynchronous, every translatable component will render twice. That's too bad for performance.